src/controllers/EvidenceUploadController.ts

```diff
@@ -164,15 +162,15 @@ export class EvidenceUploadController extends SafeNavigationBaseController<Other
                         }
                     }
 
-                    const session = request.session.extract();
+                    const session = request.session;
                     if (session != null) {
-                        const applicationData: ApplicationData = session.getExtraData()
-                            .map<ApplicationData>(data => data[APPLICATION_DATA_KEY])
-                            .orDefaultLazy(() => {
-                                const value = {} as ApplicationData;
-                                session.saveExtraData(APPLICATION_DATA_KEY, value);
-                                return value;
-                            });
+
+                        let applicationData: ApplicationData | undefined = session!.getExtraData(APPLICATION_DATA_KEY);
+
+                        if (!applicationData) {
+                            applicationData = {} as ApplicationData;
+                            session!.setExtraData(APPLICATION_DATA_KEY, applicationData);
+                        }
 
```

Comment:
this could be simplified
```suggestion
                    session.setExtraData(APPLICATION_DATA_KEY, applicationData || {} as ApplicationData);
```

Reply:
We dont need to set ExtraData if applicationData is defined